beacon-chain/forkchoice/doubly-linked-tree/forkchoice.go

```diff
@@ -172,8 +171,13 @@ func (f *ForkChoice) updateCheckpoints(ctx context.Context, jc, fc *ethpb.Checkp
 			currentSlot := slots.CurrentSlot(f.store.genesisTime)
 			if slots.SinceEpochStarts(currentSlot) < params.BeaconConfig().SafeSlotsToUpdateJustified {
 				f.store.prevJustifiedCheckpoint = f.store.justifiedCheckpoint
+				root := bytesutil.ToBytes32(jc.Root)
 				f.store.justifiedCheckpoint = &forkchoicetypes.Checkpoint{Epoch: jc.Epoch,
-					Root: bytesutil.ToBytes32(jc.Root)}
+					Root: root}
+				if err := f.updateJustifiedBalances(ctx, root); err != nil {
```

Comment:
There are three instances of `updateJustifiedBalances` in `updateCheckpoints`. Frankly, I'm a little out of touch with this function so I questioning why we need three separate instances

Reply:
They are on different branches of the conditionals. In order of appearance in this function
1. Updating only early when defending from balance attacks. 
2. Updating late if using the same mechanism. 
3. Updating when we're not using the mechanism. 

Indeed in line 214 it was missed!